File: ServiceAPI/src/main/java/com/github/pannowak/mealsadvisor/api/planning/service/PlanningService.java
package com.github.pannowak.mealsadvisor.api.planning.service;

import com.github.pannowak.mealsadvisor.api.planning.model.DayPlan;
import com.github.pannowak.mealsadvisor.api.planning.model.GroceryItem;
import reactor.core.publisher.Flux;

import java.util.Collection;

public interface PlanningService {

    Flux<DayPlan> draw(int firstDay, int lastDay);

    Flux<GroceryItem> generateShoppingList(Collection<Long> mealIds);
}

File: WebGUI/src/app/modules/planning/services/planning.service.ts
import {HttpClient, HttpParams} from '@angular/common/http';
import {Injectable} from '@angular/core';
import {DayPlan, GroceryItem} from 'app/modules/planning/models';
import {Observable} from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class PlanningService {

  constructor(private httpClient: HttpClient) { }

  private static prepareDrawQueryParams(firstDay: number, lastDay: number): HttpParams {
    return new HttpParams()
      .set('first-day', String(firstDay))
      .set('last-day', String(lastDay));
  }

  private static concatIds(mealIds: number[]): string {
    return mealIds.map(String).reduce((a, b) => `${a},${b}`);
  }

  public draw(firstDay: number, lastDay: number): Observable<DayPlan[]> {
    const queryParams = PlanningService.prepareDrawQueryParams(firstDay, lastDay);
    return this.httpClient.get<DayPlan[]>('planning/draw', {params: queryParams});
  }

  public generateShoppingList(mealIds: number[]): Observable<GroceryItem[]> {
    const queryParams = new HttpParams().set('meal-id', PlanningService.concatIds(mealIds));
    return this.httpClient.get<GroceryItem[]>('planning/shopping-list', {params: queryParams});
  }
}
